plugins/cost-insights/src/components/AlertInsights/AlertDialog.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { default as CloseIcon } from '@material-ui/icons/Close';
+import {
+  Box,
+  Button,
+  Divider,
+  Dialog,
+  DialogActions,
+  IconButton,
+  DialogContent,
+  Typography,
+} from '@material-ui/core';
+import {
+  AlertAcceptForm,
+  AlertDismissForm,
+  AlertSnoozeForm,
+} from '../../forms';
+import { useAlertDialogStyles as useStyles } from '../../utils/styles';
+import { choose } from '../../utils/alerts';
+import { Alert, AlertForm, Maybe } from '../../types';
+
+const DEFAULT_FORM_ID = 'alert-form';
+
+type AlertDialogProps = {
+  open: boolean;
+  group: string;
+  snoozed: Maybe<Alert>;
+  accepted: Maybe<Alert>;
+  dismissed: Maybe<Alert>;
+  onClose: () => void;
+  onSubmit: (data: any) => void;
+};
+
+export const AlertDialog = ({
+  open,
+  group,
+  snoozed,
+  accepted,
+  dismissed,
+  onClose,
+  onSubmit,
+}: AlertDialogProps) => {
+  const classes = useStyles();
+  const [isButtonDisabled, setDisabled] = useState(true);
+  const acceptRef = useRef<Maybe<HTMLFormElement>>(null);
+  const snoozeRef = useRef<Maybe<HTMLFormElement>>(null);
+  const dismissRef = useRef<Maybe<HTMLFormElement>>(null);
+
+  useEffect(() => {
+    if (open) {
+      setDisabled(true);
+    } else {
+      setDisabled(false);
+    }
+  }, [open]);
+
+  function disableSubmit(isDisabled: boolean) {
+    setDisabled(isDisabled);
+  }
+
+  function onDialogClose() {
+    onClose();
+    setDisabled(true);
+  }
+
+  const SnoozeForm: Maybe<AlertForm> = snoozed?.SnoozeForm ?? AlertSnoozeForm;
+  const AcceptForm: Maybe<AlertForm> = accepted?.AcceptForm ?? AlertAcceptForm;
+  const DismissForm: Maybe<AlertForm> =
+    dismissed?.DismissForm ?? AlertDismissForm;
+
+  const isSnoozingEnabled = !!snoozed?.onSnoozed;
+  const isAcceptingEnabled = !!accepted?.onAccepted;
+  const isDismissingEnabled = !!dismissed?.onDismissed;
+
+  const isSnoozeFormDisabled = snoozed?.SnoozeForm === null;
+  const isAcceptFormDisabled = accepted?.AcceptForm === null;
+  const isDismissFormDisabled = dismissed?.DismissForm === null;
+  const isFormDisabled =
+    isSnoozeFormDisabled || isAcceptFormDisabled || isDismissFormDisabled;
+
+  const status = [
+    isSnoozingEnabled,
+    isAcceptingEnabled,
+    isDismissingEnabled,
+  ] as const;
+
+  const [Action, action, actioned] =
```

Comment:
Malkovich malkovich? The case-sensitive vars make me slightly squeamish. Could remove the need for the first one with MUI `capitalize`?